catalog/src/main/java/org/killbill/billing/catalog/DefaultPlanPhase.java

```diff
@@ -75,7 +77,19 @@ public DefaultPlanPhase(final DefaultPlan parentPlan, final DefaultPlanPhase in,
         this.recurring = override != null && override.getRecurringPrice() != null ? new DefaultRecurring((DefaultRecurring) in.getRecurring(), override) : (DefaultRecurring) in.getRecurring();
         this.usages = new DefaultUsage[in.getUsages().length];
         for (int i = 0; i < in.getUsages().length; i++) {
-            usages[i] = (DefaultUsage) in.getUsages()[i];
+            final Usage curUsage = in.getUsages()[i];
+            if(override != null && override.getUsagePriceOverrides()!= null) {
+                final UsagePriceOverride usagePriceOverride = Iterables.tryFind(override.getUsagePriceOverrides(), new Predicate<UsagePriceOverride>() {
+                 @Override
+                 public boolean apply(final UsagePriceOverride input) {
```

Comment:
@sbrossie : I am validating here to see if any override exists for the current usage based on the usage name and if it does I am replacing it with the override. This would work and take care of the ordering right?